src/main/java/org/hydev/mcpm/client/arguments/parsers/SearchParser.java

```diff
@@ -30,20 +28,31 @@ public String description() {
 
     @Override
     public void configure(Subparser parser) {
-        parser.addArgument("by").choices("name", "keyword", "command")
-                .setDefault("name").dest("type")
-                .help("Specifies the Type of Search");
+        // Default search by name, if -k is specified then search by keyword
+        var type = parser.addMutuallyExclusiveGroup();
+        type.addArgument("-k", "--keyword").action(Arguments.storeTrue()).dest("byKeyword")
+            .help("Search by keyword in descriptions");
+        type.addArgument("-c", "--command").action(Arguments.storeTrue()).dest("byCommand")
+            .help("Search by command");
+
+        // Content of the search
         parser.addArgument("text").dest("text").nargs("+")
-                .help("Specifies the search text.");
+            .help("Specifies the search text.");
         parser.addArgument("-n", "--no-cache").action(Arguments.storeTrue()).dest("noCache")
-                .help("Specifies whether to use local cache or not.");
+            .help("Specifies whether to use local cache or not.");
     }
 
     @Override
     public void run(Namespace details, Consumer<String> log) {
-        var t = details.getList("text");
-        controller.searchPackages(details.getString("type"),
-                String.join(" ", details.getList("text")),
-                !details.getBoolean("noCache"), log);
+        // Convert type
+        var type = "name";
+        if (details.getBoolean("byKeyword"))
+            type = "keyword";
+        if (details.getBoolean("byCommand"))
+            type = "command";
```

Comment:
You could probably have multiple arguments with the same `dest` and different `.store*` actions to get a cleaner solution, although I don't totally trust that what I'm suggesting won't catch on fire :P